Ska/engarchive/fetch.py

```diff
@@ -563,6 +597,25 @@ def get_msid_data_from_server(h5_slice, filename):
 
         return (vals, times, bads, colnames)
 
+    @staticmethod
+    def _get_msid_data_from_maude(content, tstart, tstop, msid, unit_system):
+        """
+        Get time and values for an MSID from MAUDE.
+        Returned values are (for now) all assumed to be good.
+        """
+        out = maude.get_msids(msids=msid, start=DateTime(tstart).greta,
+                              stop=DateTime(tstop).greta)
+        out = out['data'][0] # but what about n>1 msids?
+        vals = Units(unit_system).convert(msid.upper(), out['values'])
```

Comment:
I think you can change the API for the `_get_msid_data*` methods so they just set the relevant `self` attributes like `times`, `vals`, `bads`, `colnames` within the methods instead of returning those.  I don't recall the motivation for doing it that way (probably historical) but it makes it awkward to set `_raw_vals` as suggested.